### PR TITLE
[Mypy Fix] Mypy fix for "vllm/model_executor/models/[aA][bB]"

### PR DESCRIPTION
**Pull request by yewentao256**
## Purpose

Part of https://github.com/vllm-project/vllm/issues/26533

## Test

Originally

```bash
vllm/model_executor/models/bailing_moe_mtp.py:281: error: Unexpected keyword argument "shard_id" for "default_weight_loader"  [call-arg]
vllm/model_executor/models/bailing_moe_mtp.py:351: error: Incompatible types in assignment (expression has type "str", variable has type "int")  [assignment]
Found 115 errors in 19 files (checked 824 source files)
```

Now

```bash
Run mypy for Python 3.13.................................................Passed
```

### CODE REVIEW

**Review by hmellor — CHANGES_REQUESTED** [review]
IMO there are a too many `type: ignore`s, we should really be trying to fix whatever the inconsistency is, that is the purpose of this effort

**Review by yewentao256 — COMMENTED** [review]
@hmellor Thanks for the review, updated to remove the typeignore, could you take another look? Thanks!

File: vllm/model_executor/models/interfaces.py [hunk 1]
```diff
@@ -1007,7 +998,7 @@ class SupportsQuant:
     """The interface required for all models that support quantization."""
 
     hf_to_vllm_mapper: ClassVar["WeightsMapper | None"] = None
-    packed_modules_mapping: ClassVar[dict[str, list[str]] | None] = None
+    packed_modules_mapping: ClassVar[dict[str, list[str]]] = {}
```
Comment (hmellor) [inline vllm/model_executor/models/interfaces.py]: Mutable default is not safe

File: vllm/model_executor/models/interfaces.py [hunk 2]
```diff
@@ -633,22 +633,18 @@ class SupportsPP(Protocol):
...
-        batch_size: int,
-        dtype: torch.dtype,
-        device: torch.device,
-    ) -> "IntermediateTensors":
-        """Called when PP rank > 0 for profiling purposes."""
-        ...
+    make_empty_intermediate_tensors: Callable[
+        [int, torch.dtype, torch.device], "IntermediateTensors"
+    ]
+    """Called when PP rank > 0 for profiling purposes."""
```
Comment (hmellor) [inline vllm/model_executor/models/interfaces.py]: What was the error here? I think retaining the argument names is important so this change causes us to lose information

File: vllm/model_executor/models/bailing_moe_mtp.py [hunk 3]
```diff
@@ -268,14 +269,15 @@ def load_param(
...
+            remapped_name = maybe_remap_kv_scale_name(name, params_dict)
+            if remapped_name is None:
                 return False
+            name = remapped_name
             if name not in params_dict or is_pp_missing_parameter(name, self):
                 return False
 
             param = params_dict[name]
-            weight_loader = getattr(param, "weight_loader", default_weight_loader)
+            weight_loader: Any = getattr(param, "weight_loader", default_weight_loader)
```
Comment (hmellor) [inline vllm/model_executor/models/bailing_moe_mtp.py]: `Callable`?

File: vllm/model_executor/models/bailing_moe_linear.py [hunk 4]
```diff
@@ -613,7 +618,7 @@ def load_param(name: str, tensor: torch.Tensor, shard_id=None) -> bool:
                 return False
 
             param = params_dict[name]
-            weight_loader = getattr(param, "weight_loader", default_weight_loader)
+            weight_loader: Any = getattr(param, "weight_loader", default_weight_loader)
```
Comment (hmellor) [inline vllm/model_executor/models/bailing_moe_linear.py]: `Callable`?

File: vllm/model_executor/models/adapters.py [hunk 5]
```diff
@@ -148,16 +148,19 @@ def __init__(
...
+                )
 
             # Used by SEQ_CLS_LOAD_METHODS
             self.vllm_config = vllm_config
 
             # If the model already defines a pooler instance, don't overwrite it
             pooler = getattr(self, "pooler", None)
-            if not pooler and supports_multimodal(self):
+            multimodal_model: object = self
+            if not pooler and supports_multimodal(multimodal_model):
```
Comment (hmellor) [inline vllm/model_executor/models/adapters.py:161]: IS it an option to make `supports_multimodal` do `TypeIs[SupportsMultimodal]`?

File: vllm/model_executor/models/adapters.py [hunk 6]
```diff
@@ -400,14 +409,15 @@ def verify_and_update_config(vllm_config: "VllmConfig") -> None:
...
 
 
-def _get_language_model_for_seq_cls(model) -> nn.Module:
+def _get_language_model_for_seq_cls(model: nn.Module) -> nn.Module:
     """
     Get the language model component for sequence classification conversion.
     For VLMs, returns the inner language model. For standard LLMs, returns model itself.
     """
-    if supports_multimodal(model):
+    multimodal_model: object = model
```
Comment (hmellor) [inline vllm/model_executor/models/adapters.py:452]: Same here with typeis

File: vllm/model_executor/models/interfaces.py [hunk 7]
```diff
@@ -665,20 +661,17 @@ def forward(
...
 
-    def make_empty_intermediate_tensors(
-        self,
-        batch_size: int,
-        dtype: torch.dtype,
-        device: torch.device,
-    ) -> "IntermediateTensors": ...
+    make_empty_intermediate_tensors: Callable[
+        [int, torch.dtype, torch.device], "IntermediateTensors"
+    ]
```
Comment (hmellor) [inline vllm/model_executor/models/interfaces.py]: Same here with info loss

File: vllm/model_executor/models/interfaces.py [hunk 8]
```diff
@@ -809,16 +802,14 @@ class IsHybrid(Protocol):
     def get_mamba_state_shape_from_config(
         cls,
         vllm_config: "VllmConfig",
-    ) -> tuple[tuple[int, int], tuple[int, int, int]]:
+    ) -> tuple[tuple[int, ...], ...]:
```
Comment (hmellor) [inline vllm/model_executor/models/interfaces.py]: Do these shapes vary?

File: vllm/model_executor/models/interfaces.py (hunk 1 above)
Comment (yewentao256) [reply vllm/model_executor/models/interfaces.py]: Fixed

File: vllm/model_executor/models/interfaces.py (hunk 2 above)
Comment (yewentao256) [reply vllm/model_executor/models/interfaces.py]: Cannot assign to a method [method-assign]

Comment (yewentao256) [reply vllm/model_executor/models/interfaces.py]: OK, I updated to use `_MakeEmptyIntermediateTensors`

File: vllm/model_executor/models/adapters.py (hunk 5 above)
Comment (yewentao256) [reply vllm/model_executor/models/adapters.py:161]: supports_multimodal already returns `TypeIs[SupportsMultiModal]`. Removing it produces Missing positional argument "self".

File: vllm/model_executor/models/adapters.py (hunk 6 above)
Comment (yewentao256) [reply vllm/model_executor/models/adapters.py:452]: Same above

File: vllm/model_executor/models/interfaces.py (hunk 7 above)
Comment (yewentao256) [reply vllm/model_executor/models/interfaces.py]: Fixed

File: vllm/model_executor/models/interfaces.py (hunk 8 above)
Comment (yewentao256) [reply vllm/model_executor/models/interfaces.py]: fixed

**Review by hmellor — APPROVED** [review]

File: vllm/model_executor/models/adapters.py (hunk 5 above)
Comment (hmellor) [reply vllm/model_executor/models/adapters.py:161]: Hmm, I guess mypy is not being smart enough then
